solutions/cf-1514C.mdx

```diff
@@ -0,0 +1,81 @@
+---
+id: cf-1514C
+source: CF
+title: Product 1 Modulo N
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/89810)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+
+ll gcd(ll a, ll b){
+	if(b == 0) return a;
+	return gcd(b, a % b);
+}
+
+int main(){
+	int n; cin >> n;
+	vector<ll> coprimes; 
+	/*
+	 * This array stores all of the numbers 
+	 * which are less than n and are coprime with n.
+	 */
+	for (int i = 1; i < n; i++) {
+		// A number is coprime with n if gcd(n, i) = 1
+		if (gcd(n, i) == 1) {
+			coprimes.push_back(i);
+		}
+	}
+
+	/*
+	 * The longest subsequence whose product is 1 modulo n
+	 * is the longest subarray of coprimes with a product
+	 * of 1 modulo n.
+	 */
+
+	int number_of_coprimes = coprimes.size();
+	vector<ll> prefix_mod(number_of_coprimes);
+	prefix_mod[0] = 1;
+	/*
+	 * We can take the modulo prefix to check the modulus
+	 * of the first k coprimes, where k is the end of the
+	 * subarray.
+	 */
+	for(int i = 1; i < number_of_coprimes; i++){
```

Comment:
```suggestion
	for (int i = 1; i < number_of_coprimes; i++) {
```

Do this elsewhere in the code too for consistency

Reply:
this.

Reply:
https://github.com/cpinitiative/usaco-guide/pull/1804/files#diff-85ea0144d34ffd846cba0c7700b319ae3aa340a5e0e082c0a03a9cea92eaa002R61

https://github.com/cpinitiative/usaco-guide/pull/1804/files#diff-85ea0144d34ffd846cba0c7700b319ae3aa340a5e0e082c0a03a9cea92eaa002R73

^ change these too

Reply:
done.